src/OpenTelemetry.Contrib.Shared/DiagnosticSourceInstrumentation/DiagnosticSourceSubscriber.cs

```diff
@@ -87,11 +89,11 @@ public void OnError(Exception error)
     /// <inheritdoc/>
     public void Dispose()
     {
-        this.Dispose(true);
+        this.DisposeInternal();
```

Comment:
nit: Just FYI in a sealed class you can slim down the disposal pattern. This would be fine:

```csharp
    public void Dispose()
    {
        if (Interlocked.CompareExchange(ref this.disposed, 1, 0) == 1)
        {
            return;
        }
        lock (this.listenerSubscriptions)
        {
            foreach (var listenerSubscription in this.listenerSubscriptions)
            {
                listenerSubscription?.Dispose();
            }
            this.listenerSubscriptions.Clear();
        }
        this.allSourcesSubscription?.Dispose();
        this.allSourcesSubscription = null;
    }
```